### DISCUSSION TITLE
Watch apk missing since Night Build 5th Dec

### DISCUSSION THREAD

**Discussion by JimZam** [NightscoutFoundation/xDrip#3261] [General]
Good evening. I usually unzip the apk to get the "android_wear_micro_apk.apk" and install it on mi Galaxy Watch via adb. Last time the Night Build had android wear inside it was from December the 5th. Is there any way I can get it again? Is it a bug or from now ahead it will be not available? The app in the watch didn't update by itself.
Thank you in advance

Edit: I have installed it with Wear Installer 2

**Comment by Navid200** [comment]
I have no experience with Wear OS.

Can you not get the wear component from the December 5, 2023 release?

**Comment by JimZam** [comment]
@Navid200 I can get it from December 5 release, but not from then on. But it must be somewhere inside the apk because I can install it with Wear Installer 2.
Thank you and Happy New Year :-)

**Reply by skritland** [reply]
It looks like, due to SDK version change, some packaging rules changed. I also cannot find the WearOS apk inside main apk using 7-zip. But when I extract it using apktool, then it is still in the same place.

**Reply by Npavkov** [reply]
how do I extract it using apktool???

**Reply by oakeley** [reply]
Cloning https://github.com/iBotPeaches/Apktool and building in Android studio is not a very useful solution. It is actually less work to download the source for xDrip and build THAT in Android studio. Because this then generates the wear apk in the project folder. It doesn't matter that the xDrip wear is signed with a different key you can uninstall the old wear app by clicking on the icon on the watch and then install the new self-signed apk, it still works with the official apk on the phone.

This is not a good solution

![image](https://github.com/NightscoutFoundation/xDrip/assets/228558/80c12e81-30f8-493b-a97b-e203c1736ac3)

**Comment by JimZam** [comment]
Thank you skritland, I'll.
